src/app/public/modules/country-field/country-field.component.ts

```diff
@@ -0,0 +1,295 @@
+    });
+
+    if ((<HTMLElement>elRef.nativeElement.parentElement)
+      .classList.contains('sky-phone-field-country-search')) {
+      this.isInPhoneField = true;
+    }
+  }
+
+  public ngOnInit(): void {
+    if (!this.defaultCountry) {
+      this.defaultCountry = 'us';
+    }
+
+    this.countrySearchForm.get('countrySearch').valueChanges.subscribe(newValue => {
+      if (newValue) {
+        this.selectedCountry = newValue;
+      }
+    });
+    this.addEventListeners();
+  }
+
+  public ngOnDestroy(): void {
+    this.selectedCountryChange.complete();
+    this.removeEventListeners();
+  }
+
+  /**
+   * Sets the country to validate against based on the county's iso2 code.
+   * @param countryCode The International Organization for Standardization's two-letter code
+   * for the default country.
+   */
+  public onCountrySelected(newCountry: SkyAutocompleteSelectionChange): void {
+    if (newCountry.selectedItem) {
+      this.writeValue(this.countries.find(countryInfo => countryInfo.iso2 ===
+        newCountry.selectedItem.iso2));
+    } else {
+      this.writeValue(undefined);
+    }
+  }
+
+  // Angular automatically constructs these methods.
+  /* istanbul ignore next */
+  public onChange: Function = (value: SkyCountryFieldCountry) => { };
+
+  /* istanbul ignore next */
+  public onTouched: Function = () => { };
+
+  public registerOnChange(fn: (value: any) => void): void {
+    this.onChange = fn;
+  }
+
+  public registerOnTouched(fn: () => void): void {
+    this.onTouched = fn;
+  }
+
+  // Allows Angular to disable the input.
+  public setDisabledState(disabled: boolean): void {
+    this.removeEventListeners();
+
+    if (!disabled) {
+      this.addEventListeners();
+    }
+
+    this.disabled = disabled;
+    this.changeDetector.markForCheck();
+  }
+
+  public validate(control: AbstractControl): ValidationErrors {
+    if (!this.control) {
+      this.control = control;
+    }
+    return;
+  }
+
+  public writeValue(value: SkyCountryFieldCountry): void {
+    if (!this.disabled) {
+      this.selectedCountry = value;
+    }
+  }
+
+  private addEventListeners(): void {
+    this.idle = new Subject();
+
+    const documentObj = this.windowRef.nativeWindow.document;
+    fromEvent(documentObj, 'mousedown').takeUntil(this.idle)
+      .subscribe((event: MouseEvent) => {
+        this.isInputFocused = this.elRef.nativeElement.contains(event.target);
+        this.changeDetector.markForCheck();
+      });
+
+    fromEvent(documentObj, 'focusin')
+      .takeUntil(this.idle)
+      .subscribe((event: KeyboardEvent) => {
+        this.isInputFocused = this.elRef.nativeElement.contains(event.target);
+        this.changeDetector.markForCheck();
+      });
+  }
+
+  private removeEventListeners(): void {
+    this.idle.next();
+    this.idle.complete();
+  }
+
+  private sortCountriesWithSelectedAndDefault(selectedCountry: SkyCountryFieldCountry): void {
+    let selectedCountryIndex: number;
+    let selectedCountryData;
```

Comment:
Can we give `selectedCountryData` a type?

Reply:
Done. I also added a comment as to why we are grabbing this data a second time.